src/core/api.rs

```diff
@@ -49,6 +50,40 @@ fn index() -> &'static str {
   "hello"
 }
 
+//
+#[post("/deployment/<short_sha>")]
+async fn two_factor_deployment(
+  short_sha: &str,
+  _apikey: ApiKey<'_>,
+  ctx: &State<Context>,
+) -> (Status, String) {
+  let sent_msg: serenity::model::prelude::Message = TWO_FACTOR_DEPLOYMENT_CHANNEL
+    .send_message(&ctx.http, |m| m.content("React with ✅ or ❌"))
+    .await
+    .unwrap();
+  let accept = sent_msg.react(&ctx.http, '✅').await.unwrap();
+  let reject = sent_msg.react(&ctx.http, '❌').await.unwrap();
+
+  match REACTION_COLLECTORS.try_write() {
```

Comment:
I don't like this why `try_write` ?
If you fail here the message would already be sent, and it will be very confusing for the users.

And in this case we have not many users / interactions so it's not very likely, but in general `try_write` will very ofen fail. every time another thread as the .write on it, which is very common.